src/lib/run-context/loader.ts

```diff
@@ -6,48 +6,55 @@
 import * as path from "node:path";
 
 import * as EmailLayout from "@/lib/marshal/email-layout";
+import * as MessageType from "@/lib/marshal/message-type";
+import * as Partial from "@/lib/marshal/partial";
 import * as Translation from "@/lib/marshal/translation";
 import * as Workflow from "@/lib/marshal/workflow";
 
-import { RunContext } from "./types";
+import { ResourceType, RunContext } from "./types";
+
+const buildResourceDirContext = (type: ResourceType, currDir: string) => {
+  return {
+    type,
+    key: path.basename(currDir),
+    abspath: currDir,
+    exists: true,
+  };
+};
 
 const evaluateRecursively = async (
   ctx: RunContext,
   currDir: string,
 ): Promise<RunContext> => {
-  // Check if we are inside a workflow directory, and if so update the context.
-  const isWorkflowDir = await Workflow.isWorkflowDir(currDir);
-  if (!ctx.resourceDir && isWorkflowDir) {
-    ctx.resourceDir = {
-      type: "workflow",
-      key: path.basename(currDir),
-      abspath: currDir,
-      exists: true,
-    };
-  }
+  // Check if we are inside a resource directory and if so update the context.
+  if (!ctx.resourceDir) {
+    const isWorkflowDir = await Workflow.isWorkflowDir(currDir);
+    if (isWorkflowDir) {
+      ctx.resourceDir = buildResourceDirContext("workflow", currDir);
+    }
```

Comment:
No real change except for making things a bit more DRY and simplifying the conditionals.